lib/propertyTypes.js

```diff
@@ -2,12 +2,13 @@
  * @name propertyTypes
  * @constant
  * @memberof Helpers
- * @summary Checks matching via various operators
- * @returns {Boolean} the result of the check
+ * @summary Normalizes values out to various property types
+ * @returns {Boolean|String|Number|Integer|Array}
  */
 export default {
   bool(varA) { return varA.trim().toLowerCase() === "true"; },
   float(varA) { return parseFloat(varA); },
   int(varA) { return parseInt(varA, 10); },
-  string(varA) { return varA; }
+  string(varA) { return varA; },
+  array(varA) { return varA; }
```

Comment:
I don't quite see how this would work within the context of surcharge and restriction matching rules. The assumption here (though not documented or checked, maybe should be) is that `varA` is always a string. It needs to be something that can be entered in the UI text field. And then it is run through one of these functions to make it comparable to the actual value.

One option would be to take a comma-delimited string, and this function would `.split` and then `trim` all values. But then I still think this would support only strings for the array items. So it might be best to call it `arrayOfString`, where you could also add `arrayOfBool`, `arrayOfInt`, etc. in the future.

Reply:
The code just tests any property on the product against a value so I don't see why it couldn't be an array of tags. Also there is no UI for this that I know of so I am not sure where that restriction is coming from, and if so it seems needlessly limiting

Reply:
Ok, rereading this and the code, I don't think we need this additional property type since the comparison value (tagId) is still a string. Will update.

Reply:
Removed change